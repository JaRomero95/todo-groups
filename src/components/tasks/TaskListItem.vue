<template>
  <v-list-tile>
    <v-list-tile-action>
      <v-checkbox
        v-model="task.dueComplete"
        @change="updateComplete"
      />
    </v-list-tile-action>

    <v-list-tile-content>
      <v-list-tile-title
        v-text="task.name"
        :class="{'task-done': task.dueComplete}"
        class="task-name"
      />
    </v-list-tile-content>

    <v-list-tile-action @click="deleteTask">
      <v-btn
        icon
        data-test="open-task-delete-dialog"
      >
        <v-icon>delete</v-icon>
      </v-btn>
    </v-list-tile-action>
  </v-list-tile>
</template>

<script>
import {mapActions} from 'vuex';

export default {
  props: {
    initialValue: {
      type: Object,
      required: true
    }
  },
  data() {
    return {
      task: {...this.initialValue},
    };
  },
  watch: {
    task() {
      this.task = {...this.initialValue};
    }
  },
  methods: {
    ...mapActions(['markTaskAsComplete', 'markTaskAsIncomplete']),
    deleteTask() {
      this.$emit('delete-task', this.task);
    },
    updateComplete(newValue) {
      if (newValue) {
        this.markTaskAsComplete(this.task.id);
      } else {
        this.markTaskAsIncomplete(this.task.id);
      }
    }
  }
}
</script>

<style lang="scss" scoped>
  .task-done {
    text-decoration: line-through;
  }
</style>

