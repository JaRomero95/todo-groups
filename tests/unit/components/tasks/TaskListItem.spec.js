import {shallowMount} from '@vue/test-utils';
import TaskListItem from '@/components/tasks/TaskListItem.vue';

const taskId = 19;

const options = {
  propsData: {
    initialValue: {
      id: taskId,
    }
  }
};

describe('TaskListItem', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(TaskListItem, options);
    wrapper.vm.markTaskAsComplete = jest.fn();
    wrapper.vm.markTaskAsIncomplete = jest.fn();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('updateComplete', () => {
    wrapper.vm.updateComplete(true);
    expect(wrapper.vm.markTaskAsComplete).toBeCalledWith(taskId);
  });
});
