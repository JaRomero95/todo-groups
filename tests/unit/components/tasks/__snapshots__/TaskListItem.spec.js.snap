// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListItem renders correctly 1`] = `
<v-list-tile-stub activeclass="primary--text">
  <v-list-tile-action-stub>
    <v-checkbox-stub color="accent" errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" ripple="true" valuecomparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date &amp;&amp; b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}" indeterminateicon="$vuetify.icons.checkboxIndeterminate" onicon="$vuetify.icons.checkboxOn" officon="$vuetify.icons.checkboxOff"></v-checkbox-stub>
  </v-list-tile-action-stub>
  <v-list-tile-content-stub>
    <v-list-tile-title-stub class="task-name [object Object]"></v-list-tile-title-stub>
  </v-list-tile-content-stub>
  <v-list-tile-action-stub>
    <v-btn-stub activeclass="v-btn--active" tag="button" icon="true" type="button" data-test="open-task-delete-dialog">
      <v-icon-stub>delete</v-icon-stub>
    </v-btn-stub>
  </v-list-tile-action-stub>
</v-list-tile-stub>
`;
